### DISCUSSION TITLE
Question about the problem with several sliders on one page - cannot understand?!

### DISCUSSION THREAD

**Discussion by Voodoopupp** [nolimits4web/swiper#6884] [Q&A]
Hello,

I can't quite understand the problem with multiple sliders on the same page that are supposed to use the same settings.

Examples e.g.:
https://github.com/nolimits4web/Swiper/issues/273
https://github.com/nolimits4web/swiper/issues/3797
https://github.com/nolimits4web/swiper/issues/50
https://raddy.dev/blog/multiple-instances-of-swiperjs-on-the-same-page-with-the-same-settings/

It is written again and again that there are different solutions for this: either one simply initiates Swiper with a unique CSS class each time and then defines this multiple times in javascript or one builds a loop and then dynamically initiates different classes.

I have created a simple test here, where every swiper has the same class and it still works without problems - both the pagination and the navigation arrows:
https://codepen.io/voodoopupp/pen/PoxyaXa

I just don't understand why this works for me, but the question about "multiple initiation" keeps coming up again and again?

Can someone here enlighten me in a simple way? What am I missing?

Greetings
Voodoopupp

**Comment by kirgans** [comment]
because of this :

https://github.com/nolimits4web/swiper/blob/2f61fda721232f78b1b8ed16f57953442d70384b/src/core/core.mjs#L71-L83

**Reply by Voodoopupp** [reply]
Okay, thanks for that.

Unfortunately I am not that firm at Javascript. Can you just give me a short explanation of that in words? What exactly is it doing there?

Thank you very much for your help :)

**Reply by kirgans** [reply]
It checks if there is more than 1 element with the class provided, if so it creates a new swiper instance for each elements
